### PR TITLE
About and Reps page

### PR DESCRIPTION
**Pull request by devanshgajjar**
Fixes #[5] 

**Summary**: 
Added about, reps & rep details page

**Screenshots for the change**: 

about
reps

### CODE REVIEW

**Comment by obitodarky** [comment]
Please follow pr template

**Review by obitodarky — CHANGES_REQUESTED** [review]
Please resolve merge conflicts.

**Comment by obitodarky** [comment]
@pransh15 please check the screenshots and give a feedback if this needs any changes. @harshil1712 @AyushBherwani1998 can you please review the code. Also, I think the flutter SDK version has been changed with the recent commit.

File: reps_event_app/lib/api/reps_rep_api.dart [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import 'dart:convert';
+
+import 'package:http/http.dart' as http;
+
+Future<dynamic> fetchReps({String date,String cityName}) async {
```
Comment (AyushBherwani1998) [inline reps_event_app/lib/api/reps_rep_api.dart:4]: `date` and `cityName` is only required in events

File: reps_event_app/lib/ui/about.dart [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      backgroundColor: Colors.white,
+      appBar: AppBar(
+        backgroundColor: Color(0xffd73332),
+        // title: Text("Reps Events Mobile"),
+        elevation: 0,
+      ),
+      drawer: Drawer(
```
Comment (AyushBherwani1998) [inline reps_event_app/lib/ui/about.dart]: No need of `Drawer` in the About

File: reps_event_app/lib/ui/about.dart [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+            getList(listTitle: "Events"),
+            getList(listTitle: "People")
+          ],
+        ),
+      ),
+      body: Column(
+        children: <Widget>[getHead(),
+          Padding(
+            padding: const EdgeInsets.all(16.0),
+            child: Text(
```
Comment (AyushBherwani1998) [inline reps_event_app/lib/ui/about.dart:28]: Declare a `String` for the text inside `Text`

File: reps_event_app/lib/ui/reps.dart [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      backgroundColor: Colors.white,
+      appBar: AppBar(
+        backgroundColor: Color(0xFFd73332),
+        // title: Text("Reps Events Mobile"),
+        elevation: 0,
+      ),
+      drawer: Drawer(
```
Comment (AyushBherwani1998) [inline reps_event_app/lib/ui/reps.dart]: No need of `Drawer` in Reps List Page

File: reps_event_app/lib/ui/reps.dart [hunk 5]
```diff
@@ -0,0 +1,260 @@
...
+          },
+          title: Text(
+            snapshot.data[index].fullname,
+            maxLines: 5,
+            overflow: TextOverflow.ellipsis,
+            style: TextStyle(fontFamily: 'Zilla Slab', fontSize: 18.0),
+          ),
+          subtitle:
+          Text(snapshot.data[index].city.toString()),
+          trailing: IconButton(
```
Comment (AyushBherwani1998) [inline reps_event_app/lib/ui/reps.dart]: No need of `IconButton`

File: reps_event_app/lib/ui/reps.dart [hunk 6]
```diff
@@ -0,0 +1,260 @@
...
+                      reps: snapshot.data[index],
+                    )));
+          },
+          title: Text(
+            snapshot.data[index].fullname,
+            maxLines: 5,
+            overflow: TextOverflow.ellipsis,
+            style: TextStyle(fontFamily: 'Zilla Slab', fontSize: 18.0),
+          ),
+          subtitle:
```
Comment (AyushBherwani1998) [inline reps_event_app/lib/ui/reps.dart]: No need to display the city in subtitle.
In `ListTile` we need to display Reps profile image & name

File: reps_event_app/lib/ui/splashScreen.dart [hunk 7]
```diff
@@ -47,7 +49,7 @@ class _SplashScreenPageState extends State<SplashScreenPage>
 
   navigationToHomePage() {
     Navigator.pushReplacement(
-        context, MaterialPageRoute(builder: (context) => Events()));
+        context, MaterialPageRoute(builder: (context) => Reps()));
```
Comment (AyushBherwani1998) [inline reps_event_app/lib/ui/splashScreen.dart]: `Events` was the homepage if you will navigate to `Reps` than `Reps` will be homepage.
So replace `Reps` with `Events`

**Review by AyushBherwani1998 — CHANGES_REQUESTED** [review]
Please do the required changes.

**Comment by obitodarky** [comment]
@devanshgajjar in PR description, please remove `[ ]` so that it points to the issue. If you are solving multiple issues, you can mention both issues for example: Fixes #2 and #3.

File: reps_event_app/lib/ui/reps.dart [hunk 8]
```diff
@@ -0,0 +1,260 @@
...
+      },
+    );
+  }
+
+  Future<List<RepsModel>> fetchEvent() async {
+    print("FETCH CALLED");
+    List<RepsModel> reps = [];
+    print(dateTime);
+    var response =
+    await fetchReps(date: dateTime, cityName: _searchController.text)
```
Comment (AyushBherwani1998) [inline reps_event_app/lib/ui/reps.dart]: We don't need to pass `dateTime` and `cityName` in fetchReps Api call

File: reps_event_app/lib/ui/repsDetails.dart [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+                      TextSpan(
+                        text: 'City',
+                        style: TextStyle(
+                            fontStyle: FontStyle.italic,
+                            background: Paint()
+                              ..color = Colors.red[50],
+                            fontSize: 18),
+                      ),
+                      TextSpan(
+                          text: "" +
```
Comment (obitodarky) [inline reps_event_app/lib/ui/repsDetails.dart:43]: What is this empty string `""` for?

**Review by obitodarky — COMMENTED** [review]
Can you please include a screenshot of the RepsDetails page?

File: reps_event_app/lib/ui/repsDetails.dart (hunk 9 above)
Comment (devanshgajjar) [reply reps_event_app/lib/ui/repsDetails.dart:43]: No idea

Comment (devanshgajjar) [reply reps_event_app/lib/ui/repsDetails.dart:43]: Oh wait ya, that the field name, example: "city": Vadodara , "": widget.reps.fullname

File: reps_event_app/lib/api/reps_rep_api.dart [hunk 10]
```diff
@@ -1,19 +1,6 @@
...
-  String uri = "https://reps.mozilla.org/api/v1/rep/?limit=0&offset=0";
 
-  http.Response response = await http.get(
-      uri,
-      headers: _headers);
-  print(response);
-  var responseJson = json.decode(response.body);
-  print(responseJson);
-  return (responseJson);
+
```
Comment (AyushBherwani1998) [inline reps_event_app/lib/api/reps_rep_api.dart]: Why did you remove the entire Api call logic?

File: reps_event_app/lib/ui/events.dart [hunk 11]
```diff
@@ -105,7 +105,6 @@ class _EventsState extends State<Events> {
                     scrollDirection: Axis.vertical,
                     itemCount: snapshot.data.length,
                     itemBuilder: (context, index) {
-                      ;
```
Comment (obitodarky) [inline reps_event_app/lib/ui/events.dart:111]: I don't think there should be a semi-colon here. Have you tested this on your device/emulator?

**Review by AyushBherwani1998 — CHANGES_REQUESTED** [review]
Please attach the screenshots for the recent commits of Reps List page, Reps Detail Page & About Page

File: reps_event_app/lib/api/reps_rep_api.dart (hunk 10 above)
Comment (devanshgajjar) [reply reps_event_app/lib/api/reps_rep_api.dart]: Working on it.

File: reps_event_app/lib/api/reps_rep_api.dart [hunk 12]
```diff
@@ -1,6 +1,18 @@
-
+import 'dart:convert';
+import 'package:http/http.dart' as http;
 
 Future<dynamic> fetchReps({String date,String cityName}) async {
```
Comment (obitodarky) [inline reps_event_app/lib/api/reps_rep_api.dart:4]: @devanshgajjar please resolve this issue.

File: .idea/workspace.xml [hunk 13]
```diff
@@ -18,7 +18,7 @@
         <entry file="file://$PROJECT_DIR$/reps_event_app/lib/ui/events.dart">
```
Comment (obitodarky) [inline .idea/workspace.xml:18]: Can you please explain why is the workspace.xml file being changed?

File: reps_event_app/lib/ui/events.dart (hunk 11 above)
Comment (devanshgajjar) [reply reps_event_app/lib/ui/events.dart:111]: I have removed it

**Review by obitodarky — CHANGES_REQUESTED** [review]

File: reps_event_app/lib/api/reps_rep_api.dart (hunk 12 above)
Comment (devanshgajjar) [reply reps_event_app/lib/api/reps_rep_api.dart:4]: What do I have to do here?

File: .idea/workspace.xml (hunk 13 above)
Comment (devanshgajjar) [reply .idea/workspace.xml:18]: I did not change anything in that file O.O

File: reps_event_app/lib/api/reps_rep_api.dart (hunk 12 above)
Comment (obitodarky) [reply reps_event_app/lib/api/reps_rep_api.dart:4]: We do not need `date` and `cityName` in the `fetchReps` method, this was meant for events page only. If you are taking reference of some other code snippet or file, make sure you don't include unnecessary code.

File: .idea/workspace.xml (hunk 13 above)
Comment (obitodarky) [reply .idea/workspace.xml:18]: You can check your commit log and go back to previous changes and see what caused `workspace.xml` to change.
